// react-confirm-toast/dist/cjs/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

/******************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

function __rest$1(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
}

var DefaultContext = {
  color: undefined,
  size: undefined,
  className: undefined,
  style: undefined,
  attr: undefined
};
var IconContext = React__default["default"].createContext && React__default["default"].createContext(DefaultContext);

var __assign = undefined && undefined.__assign || function () {
  __assign = Object.assign || function (t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
      s = arguments[i];

      for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
    }

    return t;
  };

  return __assign.apply(this, arguments);
};

var __rest = undefined && undefined.__rest || function (s, e) {
  var t = {};

  for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

function Tree2Element(tree) {
  return tree && tree.map(function (node, i) {
    return React__default["default"].createElement(node.tag, __assign({
      key: i
    }, node.attr), Tree2Element(node.child));
  });
}

function GenIcon(data) {
  return function (props) {
    return React__default["default"].createElement(IconBase, __assign({
      attr: __assign({}, data.attr)
    }, props), Tree2Element(data.child));
  };
}
function IconBase(props) {
  var elem = function (conf) {
    var attr = props.attr,
        size = props.size,
        title = props.title,
        svgProps = __rest(props, ["attr", "size", "title"]);

    var computedSize = size || conf.size || "1em";
    var className;
    if (conf.className) className = conf.className;
    if (props.className) className = (className ? className + ' ' : '') + props.className;
    return React__default["default"].createElement("svg", __assign({
      stroke: "currentColor",
      fill: "currentColor",
      strokeWidth: "0"
    }, conf.attr, attr, svgProps, {
      className: className,
      style: __assign(__assign({
        color: props.color || conf.color
      }, conf.style), props.style),
      height: computedSize,
      width: computedSize,
      xmlns: "http://www.w3.org/2000/svg"
    }), title && React__default["default"].createElement("title", null, title), props.children);
  };

  return IconContext !== undefined ? React__default["default"].createElement(IconContext.Consumer, null, function (conf) {
    return elem(conf);
  }) : elem(DefaultContext);
}

// THIS FILE IS AUTO GENERATED
function CgClose (props) {
  return GenIcon({"tag":"svg","attr":{"viewBox":"0 0 24 24","fill":"none"},"child":[{"tag":"path","attr":{"d":"M6.2253 4.81108C5.83477 4.42056 5.20161 4.42056 4.81108 4.81108C4.42056 5.20161 4.42056 5.83477 4.81108 6.2253L10.5858 12L4.81114 17.7747C4.42062 18.1652 4.42062 18.7984 4.81114 19.1889C5.20167 19.5794 5.83483 19.5794 6.22535 19.1889L12 13.4142L17.7747 19.1889C18.1652 19.5794 18.7984 19.5794 19.1889 19.1889C19.5794 18.7984 19.5794 18.1652 19.1889 17.7747L13.4142 12L19.189 6.2253C19.5795 5.83477 19.5795 5.20161 19.189 4.81108C18.7985 4.42056 18.1653 4.42056 17.7748 4.81108L12 10.5858L6.2253 4.81108Z","fill":"currentColor"}}]})(props);
}

function classnames(...params) {
    const dataArr = new Set();
    for (const param of params) {
        // string
        if (typeof param === 'string' && param.length > 0) {
            dataArr.add(param);
        }
        // object & array
        else if (typeof param === 'object') {
            // array
            if (Array.isArray(param)) {
                for (const e of param) {
                    if (e)
                        dataArr.add(e);
                }
            }
            // object
            else {
                for (const key in param) {
                    if (param[key]) {
                        dataArr.add(key);
                    }
                }
            }
        }
    }
    return Array.from(dataArr).join(' ');
}

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = ".ConfirmToast-module_container__HJqtg,\n.ConfirmToast-module_title_wrapper__FZVws,\n.ConfirmToast-module_buttons_container__dTerK {\n  align-items: center;\n  justify-content: center;\n  display: flex;\n}\n\n.ConfirmToast-module_light_theme__aDZlt {\n  --confirm-toast-bg: white;\n  --confirm-toast-msg-color: black;\n  --confirm-toast-icon-color: crimson;\n  --confirm-toast-btn-yes-color: seagreen;\n  --confirm-toast-btn-no-color: red;\n  --confirm-toast-btn-yes-bg: var(--confirm-toast-bg);\n  --confirm-toast-btn-no-bg: var(--confirm-toast-bg);\n  --confirm-toast-btn-hover: 0.9;\n}\n\n.ConfirmToast-module_dark_theme__j-5K8 {\n  --confirm-toast-bg: #232323;\n  --confirm-toast-msg-color: white;\n  --confirm-toast-icon-color: crimson;\n  --confirm-toast-btn-yes-color: seagreen;\n  --confirm-toast-btn-no-color: red;\n  --confirm-toast-btn-yes-bg: var(--confirm-toast-bg);\n  --confirm-toast-btn-no-bg: var(--confirm-toast-bg);\n  --confirm-toast-btn-hover: 1.4;\n}\n\n.ConfirmToast-module_snow_theme__qL2rr {\n  --confirm-toast-bg: #EAF1FD;\n  --confirm-toast-msg-color: #5077BE;\n  --confirm-toast-icon-color: #3C598E;\n  --confirm-toast-btn-yes-color: #233453;\n  --confirm-toast-btn-no-color: #233453;\n  --confirm-toast-btn-yes-bg: #D1DFFA;\n  --confirm-toast-btn-no-bg: #D1DFFA;\n  --confirm-toast-btn-hover: 1.15;\n}\n\n.ConfirmToast-module_lilac_theme__KQRB2 {\n  --confirm-toast-bg: #F5EBFA;\n  --confirm-toast-msg-color: black;\n  --confirm-toast-icon-color: #AD5BD6;\n  --confirm-toast-btn-yes-color: #9E3CCF;\n  --confirm-toast-btn-no-color: #9E3CCF;\n  --confirm-toast-btn-yes-bg: var(--confirm-toast-bg);\n  --confirm-toast-btn-no-bg: var(--confirm-toast-bg);\n  --confirm-toast-btn-hover: 1.05;\n}\n\n.ConfirmToast-module_container__HJqtg {\n  position: fixed;\n  background-color: var(--confirm-toast-bg);\n  padding: 1.2rem;\n  flex-direction: column;\n  gap: 1rem;\n  border-radius: 0.5rem;\n  user-select: none;\n  box-shadow: 0px 0px 10px 2px rgba(67, 67, 67, 0.2980392157);\n  z-index: 9999999;\n}\n\n.ConfirmToast-module_title_wrapper__FZVws {\n  gap: 0.9rem;\n}\n\n.ConfirmToast-module_title__LvF-m {\n  color: var(--confirm-toast-msg-color);\n  font-weight: bold;\n}\n\n.ConfirmToast-module_close__VIyNv {\n  color: var(--confirm-toast-icon-color);\n  font-size: 1.6rem;\n  stroke-width: 2px;\n}\n@media (hover: hover) {\n  .ConfirmToast-module_close__VIyNv:hover {\n    filter: brightness(1.3);\n  }\n}\n\n.ConfirmToast-module_buttons_container__dTerK {\n  gap: 0.6rem;\n}\n\n.ConfirmToast-module_button__CkwoM {\n  text-transform: uppercase;\n  font-weight: bold;\n  padding: 0.6rem;\n  border: none;\n  border-radius: 0.4rem;\n}\n.ConfirmToast-module_button__CkwoM.ConfirmToast-module_button_yes__3iwcr {\n  color: var(--confirm-toast-btn-yes-color);\n  background-color: var(--confirm-toast-btn-yes-bg);\n}\n.ConfirmToast-module_button__CkwoM.ConfirmToast-module_button_no__fwzkG {\n  color: var(--confirm-toast-btn-no-color);\n  background-color: var(--confirm-toast-btn-no-bg);\n}\n@media (hover: hover) {\n  .ConfirmToast-module_button__CkwoM:hover {\n    filter: brightness(var(--confirm-toast-btn-hover));\n  }\n}\n\n.ConfirmToast-module_bottom-left__2RH-x {\n  bottom: 12px;\n  left: 12px;\n  animation: ConfirmToast-module_toast-in-left__4ieJg 0.7s;\n}\n\n.ConfirmToast-module_bottom-right__KWpWO {\n  bottom: 12px;\n  right: 12px;\n  animation: ConfirmToast-module_toast-in-right__Nr-02 0.7s;\n}\n\n.ConfirmToast-module_top-left__szm9K {\n  top: 12px;\n  left: 12px;\n  animation: ConfirmToast-module_toast-in-left__4ieJg 0.7s;\n}\n\n.ConfirmToast-module_top-right__BtRgV {\n  top: 12px;\n  right: 12px;\n  animation: ConfirmToast-module_toast-in-right__Nr-02 0.7s;\n}\n\n.ConfirmToast-module_modal__4lDAi {\n  position: fixed;\n  top: 0;\n  left: 0;\n  width: 100vw;\n  height: 100vh;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  background-color: rgba(0, 0, 0, 0.5);\n  color: white;\n  transition: transform 0.3s;\n  z-index: 9999999;\n}\n\n.ConfirmToast-module_modal-content__HQS46 {\n  margin-left: auto;\n  margin-right: auto;\n  animation: ConfirmToast-module_toast-in-modal__dvlz7 0.7s;\n}\n\n@keyframes ConfirmToast-module_toast-in-right__Nr-02 {\n  from {\n    transform: translateX(100%);\n  }\n  to {\n    transform: translateX(0);\n  }\n}\n@keyframes ConfirmToast-module_toast-in-left__4ieJg {\n  from {\n    transform: translateX(-100%);\n  }\n  to {\n    transform: translateX(0);\n  }\n}\n@keyframes ConfirmToast-module_toast-in-modal__dvlz7 {\n  from {\n    transform: scale(0);\n  }\n  to {\n    transform: scale(100%);\n  }\n}";
var s = {"container":"ConfirmToast-module_container__HJqtg","title_wrapper":"ConfirmToast-module_title_wrapper__FZVws","buttons_container":"ConfirmToast-module_buttons_container__dTerK","light_theme":"ConfirmToast-module_light_theme__aDZlt","dark_theme":"ConfirmToast-module_dark_theme__j-5K8","snow_theme":"ConfirmToast-module_snow_theme__qL2rr","lilac_theme":"ConfirmToast-module_lilac_theme__KQRB2","title":"ConfirmToast-module_title__LvF-m","close":"ConfirmToast-module_close__VIyNv","button":"ConfirmToast-module_button__CkwoM","button_yes":"ConfirmToast-module_button_yes__3iwcr","button_no":"ConfirmToast-module_button_no__fwzkG","bottom-left":"ConfirmToast-module_bottom-left__2RH-x","toast-in-left":"ConfirmToast-module_toast-in-left__4ieJg","bottom-right":"ConfirmToast-module_bottom-right__KWpWO","toast-in-right":"ConfirmToast-module_toast-in-right__Nr-02","top-left":"ConfirmToast-module_top-left__szm9K","top-right":"ConfirmToast-module_top-right__BtRgV","modal":"ConfirmToast-module_modal__4lDAi","modal-content":"ConfirmToast-module_modal-content__HQS46","toast-in-modal":"ConfirmToast-module_toast-in-modal__dvlz7"};
styleInject(css_248z);

function ConfirmToast(_a) {
    var _b;
    var _c = _a.asModal, asModal = _c === void 0 ? false : _c, children = _a.children, childrenClassName = _a.childrenClassName, _d = _a.showCloseIcon, showCloseIcon = _d === void 0 ? true : _d, _e = _a.customCancel, customCancel = _e === void 0 ? 'Cancel' : _e, _f = _a.customConfirm, customConfirm = _f === void 0 ? 'Ok' : _f, customFunction = _a.customFunction, _g = _a.message, message = _g === void 0 ? 'Do you want to continue?' : _g, _h = _a.position, position = _h === void 0 ? 'bottom-right' : _h, _j = _a.theme, theme = _j === void 0 ? 'light' : _j; __rest$1(_a, ["asModal", "children", "childrenClassName", "showCloseIcon", "customCancel", "customConfirm", "customFunction", "message", "position", "theme"]);
    function Wrapper(_a) {
        var asModal = _a.asModal, children = _a.children;
        if (asModal) {
            return (React__default["default"].createElement("div", { "data-out": true, onClick: clickOutOfModal, className: s.modal }, children));
        }
        return (React__default["default"].createElement(React__default["default"].Fragment, null, children));
    }
    var _k = React.useState(false), showConfirmToast = _k[0], setShowConfirmToast = _k[1];
    var closeModal = function (e) {
        e.stopPropagation();
        setShowConfirmToast(false);
    };
    var clickOutOfModal = function (e) {
        var target = e.target;
        if (target.getAttribute('data-out')) {
            closeModal(e);
        }
    };
    var classes = classnames(s.container, (_b = {},
        _b[s[position]] = (asModal ? false : position),
        _b[s['modal-content']] = asModal,
        _b));
    return (React__default["default"].createElement("div", { onClick: function () { return setShowConfirmToast(true); }, className: classnames(s["".concat(theme, "_theme")], childrenClassName) },
        showConfirmToast &&
            React__default["default"].createElement(Wrapper, { asModal: asModal },
                React__default["default"].createElement("div", { className: classes },
                    React__default["default"].createElement("div", { className: s.title_wrapper },
                        React__default["default"].createElement("div", { className: s.title }, message),
                        showCloseIcon &&
                            React__default["default"].createElement(CgClose, { "aria-label": 'close modal', className: s.close, onClick: function (e) { closeModal(e); } })),
                    React__default["default"].createElement("div", { className: s.buttons_container },
                        React__default["default"].createElement("button", { className: "".concat(s.button, " ").concat(s.button_yes), onClick: function (e) { customFunction(); closeModal(e); } }, customConfirm),
                        React__default["default"].createElement("button", { className: "".concat(s.button, " ").concat(s.button_no), onClick: function (e) { closeModal(e); } }, customCancel)))),
        children));
}

exports.ConfirmToast = ConfirmToast;
